src/garage/torch/algos/maml.py

```diff
@@ -0,0 +1,383 @@
+            update_module_params(self._old_policy, old_theta)
+
+        return torch.stack(losses).mean()
+
+    def _compute_kl_constraint(self,
+                               itr,
+                               all_samples,
+                               all_params,
+                               set_grad=True):
+        """Compute KL divergence.
+
+        For each task, compute the KL divergence between the old policy
+        distribution and current policy distribution.
+
+        Args:
+            itr (int): Iteration number.
+            all_samples (list[list[garage.TorchTrajectoryBatch]]): Two
+                dimentional list of TorchTrajectoryBatch of size
+                [meta_batch_size * (num_grad_updates + 1)]
+            all_params (list[dict]): A list of named parameter dictionaries.
+                Each dictionary contains key value pair of names (str) and
+                parameters (torch.Tensor).
+            set_grad (bool): Whether to enable gradient calculation or not.
+
+        Returns:
+            torch.Tensor: Calculated mean value of KL divergence.
+
+        """
+        theta = dict(self.policy.named_parameters())
+        old_theta = dict(self._old_policy.named_parameters())
+
+        kls = []
+        for task_samples, task_params in zip(all_samples, all_params):
+            for i in range(self._num_grad_updates):
+                self._adapt(itr, task_samples[i], set_grad=set_grad)
+
+            update_module_params(self._old_policy, task_params)
+            with torch.set_grad_enabled(set_grad):
+                kl = self._inner_algo.compute_kl_constraint(
+                    task_samples[-1].observations)
+            kls.append(kl)
+
+            update_module_params(self.policy, theta)
+            update_module_params(self._old_policy, old_theta)
+
+        return torch.stack(kls).mean()
+
+    def _compute_loss(self, itr, batch_samples):
+        """Compute loss for one task and one gradient step.
+
+        Args:
+            itr (int): Iteration number.
+            batch_samples (garage.TorchTrajectoryBatch): Samples data for one
+                task and one gradient step.
+
+        Returns:
+            torch.Tensor: Computed loss value.
+
+        """
+        return self._inner_algo.compute_loss(itr, batch_samples)
+
+    def _compute_policy_entropy(self, task_samples):
+        """Compute policy entropy.
+
+        Args:
+            task_samples (list[garage.TorchTrajectoryBatch]): Samples data for
+                one task.
+
+        Returns:
+            torch.Tensor: Computed entropy value.
+
+        """
+        entropies = [
+            self._inner_algo.compute_policy_entropy(samples.observations)
+            for samples in task_samples
+        ]
+        return torch.stack(entropies).mean()
+
+    def _set_task(self, runner, task):
+        for env in runner._sampler._vec_env.envs:
+            env.set_task(task)
+
+    @property
```

Comment:
hmm there's not much reason to have a private property. just use the attribute directly.